PowerNSX.psm1

```diff
@@ -5789,6 +5789,44 @@ function Get-NsxManagerSystemSummary {
         $xmlsystemSummary
     }
 }
+
+function Get-NsxManagerRole {
+    <#
+    .SYNOPSIS
+    Retrieves NSX Manager Role Configuration.
+
+    .DESCRIPTION
+    The NSX Manager is the central management component of VMware NSX for
+    vSphere.
+
+    The Get-NsxManagerRole cmdlet retrieves the universal sync role of the
+    NSX Manager against which the command is run.
+
+    .EXAMPLE
+    Get-NsxManagerRole
+
+    Retreives the universal sync role from the connected NSX Manager
+    #>
+
+
+    param (
+
+        [Parameter (Mandatory=$False)]
+            #PowerNSX Connection object
+            [ValidateNotNullOrEmpty()]
+            [PSCustomObject]$Connection=$defaultNSXConnection
+
+    )
+
+    $URI = "/api/2.0/universalsync/configuration/role"
+
+    $result = invoke-nsxrestmethod -method "get" -uri $URI -connection $connection
+
+    if ( $result -is [System.Xml.XmlDocument]) {

```

Comment:
It always comes back as a XML response, but you are right in that currently it only has a single element. This could change in the future, so I've submitted the changes to cast it to XML and then use Invoke-XPathQuery to find the role element.